animate/adapt.py

```diff
@@ -111,27 +138,84 @@ def interpolate(self, f):
-        :return: its interpolation onto the adapted mesh
+        :arg f: a Function on the initial mesh
+        :type f: :class:`firedrake.function.Function`
+        :returns: its interpolation onto the adapted mesh
+        :rtype: :class:`firedrake.function.Function`
         """
         raise NotImplementedError(
             "Consistent interpolation has not yet been implemented in parallel"
         )  # TODO
 
 
-def adapt(mesh, *metrics, name=None):
+def adapt(mesh, *metrics, name=None, serialise=False, remove_checkpoints=True):
     r"""
     Adapt a mesh with respect to a metric and some adaptor parameters.
 
     If multiple metrics are provided, then they are intersected.
 
-    :param mesh: :class:`~firedrake.mesh.MeshGeometry` to be adapted.
-    :param metrics: list of :class:`.RiemannianMetric`\s
-    :param name: name for the adapted mesh
-    :return: a new :class:`~firedrake.mesh.MeshGeometry`.
+    :arg mesh: mesh to be adapted.
+    :type mesh: :class:`firedrake.mesh.MeshGeometry`
+    :arg metrics: metrics to guide the mesh adaptation
+    :type metrics: :class:`list` of :class:`.RiemannianMetric`\s
+    :kwarg name: name for the adapted mesh
+    :type name: :class:`str`
+    :kwarg serialise: if ``True``, adaptation is done in serial
+    :type serialise: :class:`bool`
+    :kwarg remove_checkpoints: if ``True``, checkpoint files are deleted after use
+    :type remove_checkpoints: :class:`bool`
+    :returns: the adapted mesh
+    :rtype: :class:`~firedrake.mesh.MeshGeometry`
     """
+
+    # Parallel adaptation is currently only supported in 3D
+    if mesh.topological_dimension() != 3:
+        serialise = False
+
+    # If already running in serial then no need to use checkpointing
+    if COMM_WORLD.size == 1:
+        serialise = False
+
+    # Combine metrics by intersection, if multiple are passed
     metric = metrics[0]
     if len(metrics) > 1:
         metric.intersect(*metrics[1:])
+
     adaptor = MetricBasedAdaptor(mesh, metric, name=name)
-    return adaptor.adapted_mesh
+    if serialise:
+        checkpoint_dir = get_checkpoint_dir()
+        if not os.path.exists(checkpoint_dir):
+            os.makedirs(checkpoint_dir)
+        metric_name, mesh_name = "tmp_metric", "tmp_adapted_mesh"
+        metric_fname = "metric_checkpoint"
+        input_fname = os.path.join(checkpoint_dir, metric_fname + ".h5")
+        output_fname = os.path.join(checkpoint_dir, "adapted_mesh_checkpoint.h5")
+
+        # In parallel, save input mesh and metric to a checkpoint file
+        save_checkpoint(metric, metric_fname, metric_name=metric_name)
+
+        # In serial, load the checkpoint, adapt and write out the result
+        if COMM_WORLD.rank == 0:
+            metric0 = load_checkpoint(metric_fname, metric_name, comm=COMM_SELF)
+            mesh0 = metric0._mesh
+            adaptor0 = MetricBasedAdaptor(
+                mesh0, metric0, name=mesh_name, comm=COMM_SELF
```

Comment:
*** Marking all connected comments with these stars ***

I guess you separated `mesh0` so that `adaptor0` would fit in a single line, but it still doesn't. In fact, it's one character too long! But if you change the `name` argument to `name=name` and don't pass `comm=COMM_SELF` (see *** comments), then this fits into one line: `adaptor0 = MetricBasedAdaptor(metric0_mesh, metric0, name=name)`

Reply:
Thanks but I don't think we should change the API just to make characters fit on a line.

Reply:
Fixed in 83d8b81.